ui/webui/src/helpers/exit.js

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright (C) 2022 Red Hat, Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with This program; If not, see <http://www.gnu.org/licenses/>.
+ */
+import cockpit from "cockpit";
+
+export const exitGui = () => {
+    const pidFile = cockpit.file("/run/anaconda/webui_script.pid", { superuser: "try", });
+    pidFile.read()
+            .then(
+                (content, tag) => {
+                    console.log("Script PID to kill:", content);
+                    cockpit.spawn(["kill", content]).then(null);
```

Comment:
cockpit.spawn also returns a promise, which you need to catch in case of errors. what you should better do is to return this cockpit.spawn here, and then add the exception catching, inside a seperate catch, so that you can catch failures from both the pidFile.read() and from the cockpit.spawn()

Reply:
Also the then(null) part seems unneeded.